File: src/main/java/core/procedures/strings/EndsWith.java
package core.procedures.strings;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

public final class EndsWith extends AFn {

  public EndsWith() {
    super(new FnArgsBuilder().min(2).max(2).mandatory(new Class[]{CharSequence.class, CharSequence.class}).build());
  }

  @Override
  public String getName() {
    return "ends-with?";
  }

  @Override
  public Boolean apply2(Object arg1, Object arg2) {
    return arg1.toString().endsWith(arg2.toString());
  }
}

File: src/main/java/core/procedures/strings/EndsWith.kt
package core.procedures.strings

import core.procedures.AFn
import core.procedures.Arity.Exactly

class EndsWith : AFn<CharSequence?, Boolean>(name = "ends-with?", isPure = true, arity = Exactly(2),
                                             mandatoryArgsTypes = arrayOf(CharSequence::class.java,
                                                                          CharSequence::class.java)) {

    override operator fun invoke(arg1: CharSequence?, arg2: CharSequence?) = arg1!!.endsWith(arg2!!)
}
